resources/assets/lib/beatmaps/search-filter.tsx

```diff
@@ -0,0 +1,113 @@
+// Copyright (c) ppy Pty Ltd <contact@ppy.sh>. Licensed under the GNU Affero General Public License v3.0.
+// See the LICENCE file in the repository root for full licence text.
+
+import { FilterKey } from 'beatmapset-search-filters';
+import { action, computed, makeObservable } from 'mobx';
+import { observer } from 'mobx-react';
+import core from 'osu-core-singleton';
+import * as React from 'react';
+import { classWithModifiers } from 'utils/css';
+import { formatNumber } from 'utils/html';
+import { FilterOption } from './available-filters';
+
+interface Props {
+  multiselect: boolean;
+  name: FilterKey;
+  options: FilterOption[];
+  title?: string;
+}
+
+@observer
+export class SearchFilter extends React.PureComponent<Props> {
```

Comment:
forgot about this one but this probably shouldn't inherit PureComponent

Reply:
still need update on this, I think 👀

Reply:
weird, I though I did; maybe it was a different branch 🤔